### ISSUE TITLE
leaflet dependency issue 0.7.3

### ISSUE THREAD

**Issue by phpmaps** [Esri/esri-leaflet#636]
Looks like there is a leaflet dependency issue that impacts the master branch.  The readme and the [release notes](https://github.com/Esri/esri-leaflet/releases/tag/v1.0.0) say there is support for 0.7.3.  However, the [build uses the leaflet beta version](https://github.com/Esri/esri-leaflet/blob/master/package.json#L14 ).  When switching to 0.7.3  there are exceptions saying  ```Cannot read property 'extend' of undefined```.

To fix this perhaps require Leaflet Beta 1?

```
<!--
This will work
<link rel="stylesheet" href="http://cdn.leafletjs.com/leaflet-1.0.0-b1/leaflet.css" />
<script src="http://cdn.leafletjs.com/leaflet-1.0.0-b1/leaflet.js"></script>
-->

<!-- This will not work -->
<link rel="stylesheet" href="//cdn.jsdelivr.net/leaflet/0.7.3/leaflet.css" />
<script src="//cdn.jsdelivr.net/leaflet/0.7.3/leaflet.js"></script>
```

**Comment by jgravois** [comment]
version 1.0.0 of esri-leaflet corresponds with Leaflet 0.7.3.  version 2.0.0 and above correspond with Leaflet 1.0.0.

you can find more information [here](https://github.com/Esri/esri-leaflet/wiki/Esri-Leaflet-1.0.0-Announcement)

the sample [app](https://github.com/Esri/esri-leaflet/blob/master/debug/sample.html) included in the repo loads Leaflet from the node dependencies of the project.  

it's a little hard to sift through the requirements in the change log, because we 
please let me know what else I can do to make the compatibility requirements more clear.

**Comment by phpmaps** [comment]
@jgravois I think I assumed that the master branch is at 0.7.3.  To help me understand can you let me what version is associated with the master branch?  My experience was when I cloned master and created an app w/ leaflet 0.7.3, the esri-leaflet plugin at ```master``` hit the exception ```Cannot read property 'extend' of undefined```.  

So are you saying that is known issue until #620 gets merged?  Let me know, so I can better digest what's going on in #620.

**Comment by jgravois** [comment]
> I assumed that the master branch is at `0.7.3`

nope, `master` is our live code (only compatible with leaflet 1.0.0b1, next tag will be either 2.0.0beta5 or possibly just plain old 2.0.0).

the only thing going on in #620 is changes to our documentation.  when i merge into `master` and sync `gh-pages`, the [live website](http://esri.github.io/esri-leaflet) will show Leaflet 1.0.0b1 compatible sample apps.

feel free to post a jsbin or equivalent repro case if you need help debugging.

**Comment by phpmaps** [comment]
Great - I guess you are saying that this issue will be resolved when #620 gets merged.  When I logged it it was really about the master's readme saying the leaflet dependency is Leaflet 0.7.3 when it's really Leaflet 1.0.0b1.  @jgravois would it be ok that I make a pull request now with that simple read me change?

**Comment by jgravois** [comment]
> would it be ok that I make a pull request now with that simple read me change?

please do.

it would also be really helpful if you also bumped the CDN versions in the inline sample and explicity mentioned that version `0.7.3` of Leaflet is supported when using version `1.0.0` of esri-leaflet **and** version `1.0.0.b1` of Leaflet is supported (with Esri Leaflet `2.x`).

**Comment by andygup** [comment]
Can you also consider updating the [roadmap](https://github.com/Esri/esri-leaflet/wiki/Roadmap)? I understand this may need to be a separate issue, but it is related.

**Comment by jgravois** [comment]
> Can you also consider updating the roadmap?

@andygup.  good call.  i just made a few edits to reflect what's changed.

**Comment by andygup** [comment]
@jgravois perfect, thanks! That's in sync with what you and I chatted about, and I'll be including a link to the page in my prezo next week.

**Comment by jgravois** [comment]
closed via 491791833344686a8c2a68b79c0e140b9ae733a3
